### PR TITLE
Bump all dependencies + fixes for new syn/quote

### PR DESCRIPTION
**Pull request by jonhoo**
This updates all dependencies to their newer versions, including fixes for the relatively large changes to the upstream `syn` and `quote` crates.

### CODE REVIEW

File: zookeeper-derive/Cargo.toml [hunk 1]
```diff
@@ -9,5 +9,6 @@ description = "Derive implementations used to build the ZooKeeper library"
 proc-macro = true
 
 [dependencies]
-quote = "~0.3.15"
-syn = "~0.11.11"
+quote = "0.6.3"
+syn = "0.14.4"
+#proc-macro2 = { version = "0.4", features = ["nightly"] }
```
Comment (bonifaido) [inline zookeeper-derive/Cargo.toml]: Do you need this comment snippet during testing or something else?

Comment (bonifaido) [reply zookeeper-derive/Cargo.toml]: Other than that looks good to me. Thank you!

**Comment by jonhoo** [comment]
I left the comment there because for debugging it can be useful to switch to `proc-macro2`, as it gives you better error messages :)

**Comment by bonifaido** [comment]
Ah, great! (I haven't used it before). I did cut a new release (0.5.3) with your changes included.

**Comment by jonhoo** [comment]
@bonifaido I think you may also need to cut a new release of `zookeeper_derive`. The `version` argument included [in `/Cargo.toml`](https://github.com/bonifaido/rust-zookeeper/blob/master/Cargo.toml#L21) seems to cause `cargo` to use `zookeeper_derive` from `crates.io` rather than the one included with `zookeeper`, which uses the old `syn` and `quote` dependencies.

**Pull request merged**